packages/core/adder/preconditions.ts

```diff
@@ -40,6 +43,29 @@ function getGlobalPreconditions(
                     }
                 },
             },
+            {
+                name: "supported environments",
+                // eslint-disable-next-line @typescript-eslint/require-await
+                run: async () => {
+                    const addersForInvalidEnvironment = adderDetails.filter((x) => {
+                        const supportedEnvironments = x.config.metadata.environments;
+                        if (projectType == "kit" && !supportedEnvironments.kit) return true;
+                        if (projectType == "svelte" && !supportedEnvironments.svelte) return true;
+
+                        return false;
+                    });
+
+                    if (addersForInvalidEnvironment.length == 0) {
+                        return { success: true, message: undefined };
+                    }
+
+                    const messages: string[] = [];
+                    for (const adder of addersForInvalidEnvironment) {
+                        messages.push(`"${adder.config.metadata.name}" does not support "${projectType}"`);
+                    }
```

Comment:
another nit, though it's a matter of preference:
```suggestion
                    const messages = addersForInvalidEnvironment.map(
                        (adder) => `"${adder.config.metadata.name}" does not support "${projectType}"`,
                    );
```

(not sure why the suggestion on github is so rough with it on the formatting 🤔 )